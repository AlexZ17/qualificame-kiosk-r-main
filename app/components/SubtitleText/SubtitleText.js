import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';

import styles from './SubtitleTextStyle';
import { Fonts, Colors } from 'qualificame-kiosk/app/styles';

export default class SubtitleText extends PureComponent {
	render() {
		const textColor = { color: Colors[this.props.color] };
		const weight = { fontWeight: Fonts.weight[this.props.weight] };
		const align = { textAlign: this.props.align };
		
		return (
			<View>
				<Text style={[styles.subtitleText, textColor, weight, align]}>
					{this.props.text}
				</Text>
			</View>
		);
	}
}

SubtitleText.propTypes = {
	text: PropTypes.string,
	color: PropTypes.string,
	weight: PropTypes.string,
	align: PropTypes.string,
}

SubtitleText.defaultProps = {
	text: '',
	color: "black",
	weight: "regular",
	align: "left"
}