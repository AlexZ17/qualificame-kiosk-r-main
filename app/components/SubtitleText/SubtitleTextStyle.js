import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-kiosk/app/styles';

export default StyleSheet.create({
	subtitleText: {
		fontSize: Fonts.size.subtitle
	}
});
