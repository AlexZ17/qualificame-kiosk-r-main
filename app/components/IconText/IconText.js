import React, { PureComponent } from 'react';
import { Text, View, Platform } from 'react-native';
import PropTypes from 'prop-types';

import styles from './IconTextStyle';
import SubtitleText from 'qualificame-kiosk/app/components/SubtitleText';
import { Feather } from '@expo/vector-icons';
import { Colors } from 'qualificame-kiosk/app/styles';

export default class IconText extends PureComponent {
	render() {
		let color = {color: Colors[this.props.color]}
		return (
			<View style={[styles.row, styles.alignItemsCenter, (this.props.AddVerticalMargin && styles.baseVerticalMargin), (this.props.AddHorizontalMargin && styles.baseHorizontalMargin)]}>
				<Feather
					name={this.props.iconName} 
					size={Platform.OS === 'ios' ? 29:24} 
					style={[color]}
				/>
				<View style={styles.smallHorizontalMargin}>
					<SubtitleText
						text={this.props.text}
						color={this.props.color}
					/>
				</View>
			</View>
		);
	}
}

	IconText.propTypes = {
		iconName: PropTypes.string,
		text: PropTypes.string,
		color: PropTypes.string,
		AddVerticalMargin: PropTypes.bool,
		AddHorizontalMargin: PropTypes.bool,
	}

	IconText.defaultProps = {
		iconName: 'phone',
		text: '',
		color: 'white',
		AddVerticalMargin: false,
		AddHorizontalMargin: false,
	}
