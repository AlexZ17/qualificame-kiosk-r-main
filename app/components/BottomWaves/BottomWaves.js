import React, { PureComponent } from 'react';
import {
	ImageBackground,
	View
} from 'react-native';
import PropTypes from 'prop-types';

import TitleText from 'qualificame-kiosk/app/components/TitleText'; 
import SubtitleText from 'qualificame-kiosk/app/components/SubtitleText'; 
import IconText from 'qualificame-kiosk/app/components/IconText'; 

import styles from './BottomWavesStyle';

export default class BottomWaves extends PureComponent {
	render() {
		let imgSrc;
		switch(this.props.color) {
			case 'green':
				imgSrc = require('qualificame-kiosk/assets/waves-green.png')
				break;
			case 'pink':
				imgSrc = require('qualificame-kiosk/assets/waves-pink.png')
				break;
			case 'blue':
				imgSrc = require('qualificame-kiosk/assets/waves-blue.png')
				break;
			case 'purple':
				imgSrc = require('qualificame-kiosk/assets/waves-purple.png')
				break;
			case 'orange':
				imgSrc = require('qualificame-kiosk/assets/waves-orange.png')
				break;
			default:
				imgSrc = require('qualificame-kiosk/assets/waves-green.png')
				break;

		}
		return (
			<React.Fragment>
				<ImageBackground
					source={imgSrc}
					style={[
						styles.absolute,
						styles.setToBot,
						styles.behind,
						styles.footerImage,
						styles.justifyContentFlexEnd,
						styles.alignItemsCenter,
						styles.doubleBaseBottomPadding
					]}
					resizeMode='stretch'
				>
					{this.props.showData && 
						this.props.isBad ? (
							<View style={this.props.isBad && [
								styles.row, styles.fullWidth, styles.baseHorizontalPadding,styles.justifyContentSpaceAround]}>
								<View style={styles.flex1}>
									<TitleText
										color='white'
										text='Tu opinión es muy importante para nosotros'
										align="right"
									/>
									<SubtitleText
										color='white'
										text='¡Ayúdanos a mejorar! Comúnicate a:'
										align="right"
									/>
								</View>
								<View style={[styles.baseLeftMargin, styles.row, styles.smallTopMargin, styles.flex1]}>
									<IconText
										text={this.props.phone}
										iconName='phone'
										color='white'
										AddHorizontalMargin
									/>
									<IconText
										text={this.props.email}
										iconName='mail'
										color='white'
										AddHorizontalMargin
									/>
								</View>
							</View> 
						) : (
							<View>
								<TitleText
									color='white'
									text='Tu opinión es muy importante para nosotros'
									align="center"
								/>
								<View style={[styles.row, styles.smallTopMargin, styles.justifyContentSpaceAround]}>
									<IconText
										text={this.props.phone}
										iconName='phone'
										color='white'
										AddHorizontalMargin
									/>
									<IconText
										text={this.props.email}
										iconName='mail'
										color='white'
										AddHorizontalMargin
									/>
								</View>
							</View> 
						)
					}
				</ImageBackground>
			</React.Fragment>
		);
	}
}

BottomWaves.propTypes = {
	showData: PropTypes.bool,
	email: PropTypes.string,
	phone: PropTypes.string
}

BottomWaves.defaultProps = {
	showData: true,
	email: '',
	phone: ''
}
