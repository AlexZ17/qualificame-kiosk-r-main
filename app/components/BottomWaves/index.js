import BottomWaves from './BottomWaves';

export default BottomWaves;
