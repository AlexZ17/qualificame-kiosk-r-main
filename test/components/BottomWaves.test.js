import React from 'react';
import { shallow } from 'enzyme';

import { ImageBackground } from 'react-native';

import BottomWaves from 'qualificame-kiosk/app/components/BottomWaves';
import IconText from 'qualificame-kiosk/app/components/IconText';
import SubtitleText from 'qualificame-kiosk/app/components/SubtitleText';

import wavesImg from 'qualificame-kiosk/assets/waves-green.png';

describe('rendering', () => {
	const wrapper = shallow(<BottomWaves />);
	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('renders IconText components', () => {
		expect(wrapper.find(IconText)).toHaveLength(2);
	});
	it('renders SubtitleText component', () => {
		expect(wrapper.find(SubtitleText)).toHaveLength(1);
	});
	it('renders image', () => {
		expect(wrapper.find(ImageBackground).prop('source')).toEqual(wavesImg);
	})
})