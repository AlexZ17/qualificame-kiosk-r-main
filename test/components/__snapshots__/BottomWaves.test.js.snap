// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Fragment>
  <ImageBackground
    resizeMode="stretch"
    source={1}
    style={
      Array [
        Object {
          "position": "absolute",
        },
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
        },
        Object {
          "zIndex": -1,
        },
        Object {
          "height": 187.5,
          "width": 1334,
        },
        Object {
          "justifyContent": "flex-end",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "paddingBottom": 16,
        },
      ]
    }
  >
    <Component>
      <SubtitleText
        align="center"
        color="white"
        text="Tu opiniÃ³n es muy importante para nosotros"
        weight=""
      />
      <Component
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <IconText
          AddHorizontalMargin={true}
          AddVerticalMargin={false}
          color="white"
          iconName="phone"
          text="771 428 9913"
        />
        <IconText
          AddHorizontalMargin={true}
          AddVerticalMargin={false}
          color="white"
          iconName="mail"
          text="hola@plazasanjuan.com"
        />
      </Component>
    </Component>
  </ImageBackground>
</Fragment>
`;
